test/onnx/test_pytorch_jit_onnx.py

```diff
@@ -0,0 +1,56 @@
+# Owner(s): ["module: onnx"]
+import torch
+import onnxruntime
+import unittest
+from test_pytorch_onnx_onnxruntime import run_ort, ort_compare_with_pytorch
+from torch._C import parse_ir
+
+
+class _TestJITIRToONNX:
+    """Abstract base class for test cases.
+
+    Intentionally not a sub-class of unittest.TestCase so that unittest / pytest
+    don't run it directly. unitest.TestCase is mixed in as another base class when
+    creating concrete sub-types. See MakeTestCase().
+    """
+    opset_version = -1  # Sub-classes must override
+    ort_providers = ["CPUExecutionProvider"]
+
+    def run_test(self, graph_ir, example_inputs):
+        graph = parse_ir(graph_ir)
+        jit_outs = torch._C._jit_interpret_graph(graph, example_inputs)
+
+        f = torch.onnx.utils._export_jit_graph_to_onnx_model_proto(
+            graph,
+            torch.onnx.OperatorExportTypes.ONNX)
+        ort_sess = onnxruntime.InferenceSession(f, providers=self.ort_providers)
+        ort_outs = run_ort(ort_sess, example_inputs)
+
+        ort_compare_with_pytorch(ort_outs, jit_outs, rtol=1e-3, atol=1e-7)
+
+    def test_example_ir(self):
+        graph_ir = """
+        graph(%1 : Float(2, 3),
+              %2 : Float(2, 3)):
+          %3 : int = prim::Constant[value=1]()
+          %4 : Float(2, 3) = aten::add(%1, %2, %3)
+          return (%4)
+        """
+        a = torch.randn(2, 3)
+        b = torch.randn(2, 3)
+        self.run_test(graph_ir, (a, b))
+
+
+def MakeTestCase(opset_version: int) -> type:
+    name = f"TestJITIRToONNX_opset{opset_version}"
+    return type(str(name),
+                (unittest.TestCase,),
+                dict(_TestJITIRToONNX.__dict__,
```

Comment:
what is the purpose of this dict?

Reply:
creating unittest.TestCase object with test cases defined in _TestJITIRToONNX.